src/snapshotcloneserver/clone/clone_service_manager.cpp

```diff
@@ -231,6 +232,25 @@ int CloneServiceManager::Flatten(
 
     auto cloneInfoMetric = std::make_shared<CloneInfoMetric>(taskId);
     auto closure = std::make_shared<CloneClosure>();
```

Comment:
not only Flatten has this bug,  'not lazy clone' also should fix.

Reply:
> not only Flatten has this bug, 'not lazy clone' also should fix.

The file is invisible before not lazy clone task finished, so it will not meet with discard.